todo_app/tests_e2e/test_selenium.py

```diff
@@ -0,0 +1,360 @@
+    done_list_table_cell = driver.find_element_by_xpath("//table[@id='done-table']/tbody/tr[1]/td[1]")
+
+    #make sure there is one row in done table and none in the todo and doing
+    assert len(to_do_list_table_rows) == 0
+    assert len(doing_list_table_rows) == 0
+    assert len(done_list_table_rows) == 1
+    #make sure the done table row has a value and the text is "New Selenium Test Item"
+    assert done_list_table_cell.text == new_card_name
+
+
+def test_can_create_new_to_do_item_start_it_complete_it_and_restart_it(driver, test_app):
+    new_card_name = "New Selenium Test Item"
+    driver.get('http://localhost:5000/')
+    new_to_do_textbox = driver.find_element_by_name("todoTitle")
+    new_to_do_textbox.send_keys(new_card_name)
+    save_button = driver.find_element_by_id("save_new")
+    ActionChains(driver).click(save_button).perform()
+
+    element_present = expected_conditions.presence_of_element_located((By.XPATH, "//table[@id='to-do-table']/tbody/tr[1]"))
+    WebDriverWait(driver, 60).until(element_present)
+
+    to_do_list_table_rows = driver.find_element_by_xpath("//table[@id='to-do-table']/tbody").find_elements_by_tag_name("tr")
+    doing_list_table_rows = driver.find_element_by_xpath("//table[@id='doing-table']/tbody").find_elements_by_tag_name("tr")
+    done_list_table_rows = driver.find_element_by_xpath("//table[@id='done-table']/tbody").find_elements_by_tag_name("tr")
+    to_do_list_table_cell = driver.find_element_by_xpath("//table[@id='to-do-table']/tbody/tr[1]/td[1]")
+
+    #make sure there is one row in todo table and none in the doing and done
+    assert len(to_do_list_table_rows) == 1
+    assert len(doing_list_table_rows) == 0
+    assert len(done_list_table_rows) == 0
+    #make sure the todo table row has a value and the text is "New Selenium Test Item"
+    assert to_do_list_table_cell.text == new_card_name
+    
+    doing_button = driver.find_element_by_xpath("//table[@id='to-do-table']/tbody/tr[1]/td[2]/a[text()='Doing']")
+    ActionChains(driver).click(doing_button).perform()
+
+    element_present = expected_conditions.presence_of_element_located((By.XPATH, "//table[@id='doing-table']/tbody/tr[1]"))
+    WebDriverWait(driver, 60).until(element_present)
+
+    to_do_list_table_rows = driver.find_element_by_xpath("//table[@id='to-do-table']/tbody").find_elements_by_tag_name("tr")
+    doing_list_table_rows = driver.find_element_by_xpath("//table[@id='doing-table']/tbody").find_elements_by_tag_name("tr")
+    done_list_table_rows = driver.find_element_by_xpath("//table[@id='done-table']/tbody").find_elements_by_tag_name("tr")
+    doing_list_table_cell = driver.find_element_by_xpath("//table[@id='doing-table']/tbody/tr[1]/td[1]")
+
+    #make sure there is one row in doing table and none in the todo and done
+    assert len(to_do_list_table_rows) == 0
+    assert len(doing_list_table_rows) == 1
+    assert len(done_list_table_rows) == 0
```

Comment:
You repeat the lines to get the table rows and assert the three lengths many times in this file, so it's probably worth writing a function that does that to avoid repetition.